src/ChipArrayField/ChipArrayField.jsx

```diff
@@ -0,0 +1,209 @@
+
+  return {
+    root: {
+      position: 'relative',
+      borderRadius: 4,
+      fontSize: 16,
+      backgroundColor: theme.palette.common.white,
+      border: `1px solid ${borderColor}`,
+      width: 'auto',
+      transition: theme.transitions.create('border-color', {
+        duration: theme.transitions.duration.shorter
+      }),
+      '&:focus': {
+        borderRadius: 4,
+        borderColor: '#80bdff',
+        boxShadow: '0 0 0 0 rgba(255,255,255,.25)'
+      },
+      '&:hover': {
+        borderRadius: 4,
+        borderColor: '#80bdff',
+        boxShadow: '0 0 0 0 rgba(255,255,255,.25)'
+      },
+      marginTop: '24px'
+    },
+    error: {
+      borderColor: theme.palette.error.main
+    },
+    input: {
+      marginTop: 4,
+      padding: '10px 12px'
+    },
+    chip: {
+      margin: '8px'
+    },
+    label: {
+      top: -12
+    }
+  }
+}
+
+const ChipRenderer = ({ value, isFocused, isDisabled, handleClick, handleDelete, defaultStyle }, key) => (
+  <Chip
+    key={key}
+    style={{
+      ...defaultStyle,
+      pointerEvents: isDisabled ? 'none' : undefined,
+      backgroundColor: value.valid ? blue[300] : grey[300],
+      float: 'left',
+      margin: '0 8px 8px 0'
+    }}
+    onClick={handleClick}
+    onDelete={handleDelete}
+    label={value.value}
+  />
+)
+
+export const StyledChipInput = withStyles(styles)(ChipInput)
+
+/**
+ * The `<ChipArrayField>` component allows the editing and validation of a
+ * collection of items.
+ *
+ * It is designed to be a better experience when things like email addresses by
+ * preventing the need to comma separate the addresses manually.
+ *
+ * It is under the hood of EmailArrayField, but without some email specific features
+ *
+ * ~~~js
+ * import { ChipArrayField } from '@theconversation/ui'
+ *
+ * <ChipArrayField entries={["First"]} />
+ * ~~~
+ */
+class ChipArrayField extends React.Component {
+  constructor () {
+    super()
+    this.state = {
+      entries: []
+    }
+    this.handleAddEntry = this.handleAddEntry.bind(this)
+    this.handleDeleteEntry = this.handleDeleteEntry.bind(this)
+    this.sortEntries = this.sortEntries.bind(this)
+    this.validateEntries = this.validateEntries.bind(this)
+  }
+
+  componentDidMount () {
+    const { entries } = this.props
+    this.setState({
+      entries
+    })
+  }
+
+  handleAddEntry (value) {
+    let { entries } = this.state
+    entries.push(value)
+    this.setState({ entries })
+  }
+
+  handleDeleteEntry (entry, index) {
+    let { entries } = this.state
+    entries.splice(index, 1)
+    this.setState({ entries: entries })
+  }
+
+  sortEntries (entries) {
+    if (this.props.onSortEntry) {
+      return entries.concat().sort(this.props.onSortEntry)
+    }
+    return entries.concat().sort((a, b) => {
+      const valA = a.value.toLowerCase()
+      const valB = b.value.toLowerCase()
+      return (valA < valB) ? -1 : (valA > valB) ? 1 : 0
+    })
+  }
```

Comment:
`entries` prop is an Array. Why the `concat()` empty call?

Reply:
Concat copies the array so that the sort doesn't mutate the existing array.

Reply:
Always forget that there are more than X ways of duplicating arrays on JS.